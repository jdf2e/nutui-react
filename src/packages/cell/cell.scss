.nut-cell {
  position: relative;
  display: flex;
  flex-direction: row;
  width: 100%;
  line-height: $cell-line-height;
  padding: $cell-padding;
  background-color: $cell-background-color;
  border-radius: $cell-border-radius;
  box-shadow: $cell-box-shadow;
  font-size: $cell-title-font-size;
  color: $cell-title-color;
  margin-bottom: 10px;
  box-sizing: border-box;

  &-group-item {
    border-radius: 0;
    box-shadow: 0 0 transparent;
    margin: 0;
  }

  &-left {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    flex: 1;
  }

  &-description {
    font-size: $cell-description-font-size;
    color: $cell-description-color;
  }

  &-extra {
    display: flex;
    flex-direction: row;
    justify-content: flex-end;
    align-items: center;
    flex: 1;
    flex-shrink: 0;
    min-width: 0;
    word-break: break-all;
    font-size: $cell-extra-font-size;
    color: $cell-extra-color;
  }

  &-divider {
    display: flex;
    min-height: 1px;
    padding-left: $cell-divider-left;
    padding-right: $cell-divider-right;
    &-inner {
      display: flex;
      height: 1px;
      width: 100%;
      border-top: $cell-divider-border-bottom;
    }
    &-rtl {
      padding-left: $cell-divider-right;
      padding-right: $cell-divider-left;
    }
  }
}
