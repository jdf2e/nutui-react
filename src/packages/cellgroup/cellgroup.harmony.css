.nut-cell-group {
  display: block;
}
.nut-cell-group-title {
  display: inherit;
  padding: 0 10px;
  color: #1a1a1a;
  font-size: 14px;
  line-height: 20px;
  margin-top: 30px;
  margin-bottom: 10px;
}
.nut-cell-group-description {
  display: inherit;
  padding: 0 10px;
  color: #595959;
  font-size: 12px;
  line-height: 16px;
  margin-top: 10px;
  margin-bottom: 10px;
}
.nut-cell-group-wrap {
  display: inherit;
  border-radius: 6px;
  overflow: hidden;
  background-color: #ffffff;
  margin: 10px 0;
}
.nut-cell-group-wrap .nut-cell {
  margin: 0;
  box-shadow: none;
  border-radius: 0;
}
.nut-cell-group-wrap-divider .nut-cell-divider {
  position: absolute;
  box-sizing: border-box;
  pointer-events: none;
  right: 16px;
  bottom: 0;
  left: 16px;
  transform: scaleY(0.5);
  border-top: 2px solid rgba(0, 0, 0, 0.06);
}
.nut-cell-group-wrap-divider .nut-cell:last-child .nut-cell-divider {
  border-top: 0 !important;
}

[dir=rtl] .nut-cell-group-wrap-divider .nut-cell-divider {
  left: 16px;
  right: 16px;
}

.nut-rtl .nut-cell-group-wrap-divider .nut-cell-divider {
  left: 16px;
  right: 16px;
}