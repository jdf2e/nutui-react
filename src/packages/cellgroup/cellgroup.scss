.nut-cell-group {
  display: block;
  &__title {
    display: inherit;
    padding: $cell-group-title-padding;
    color: $cell-group-title-color;
    font-size: $cell-group-title-font-size;
    line-height: $cell-group-title-line-height;
    margin-top: 30px;
    margin-bottom: 10px;
  }
  &__desc {
    display: inherit;
    padding: $cell-group-desc-padding;
    color: $cell-group-desc-color;
    font-size: $cell-group-desc-font-size;
    line-height: $cell-group-desc-line-height;
    margin-top: 10px;
    margin-bottom: 10px;
  }
  &__wrap {
    display: inherit;
    border-radius: $cell-border-radius;
    overflow: hidden;
    background-color: $cell-group-background-color;
    margin: 10px 0;
    .nut-cell {
      margin: 0;
      box-shadow: none;
      border-radius: 0;
    }
  }
  .nut-cell {
    &::after {
      border-bottom: $cell-after-border-bottom;
    }
  }
}
