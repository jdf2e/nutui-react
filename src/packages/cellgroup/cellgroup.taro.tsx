import React, { FunctionComponent, ReactNode } from 'react'

import classNames from 'classnames'
import { BasicComponent, ComponentDefaults } from '@/utils/typings'

export interface CellGroupProps extends BasicComponent {
  title: ReactNode
  description: ReactNode
  children?: ReactNode
  divider: boolean
}
const defaultProps = {
  ...ComponentDefaults,
  title: '',
  description: '',
  divider: true,
} as CellGroupProps

const classPrefix = 'nut-cell-group'

export const CellGroup: FunctionComponent<Partial<CellGroupProps>> = (
  props
) => {
  const { children, className, style, title, description, divider, ...rest } = {
    ...defaultProps,
    ...props,
  }
  return (
    <div className={classNames(classPrefix, className)} {...rest}>
      {title ? <div className={`${classPrefix}__title`}>{title}</div> : null}
      {description ? (
        <div className={`${classPrefix}__description`}>{description}</div>
      ) : null}
      <div
        className={`${classPrefix}__wrap ${
          divider ? `${classPrefix}__wrap--divider` : ''
        }`}
      >
        {children}
      </div>
    </div>
  )
}

CellGroup.defaultProps = defaultProps
CellGroup.displayName = 'NutCellGroup'
