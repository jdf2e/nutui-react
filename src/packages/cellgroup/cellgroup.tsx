import React, { FunctionComponent, ReactNode } from 'react'

import bem from '@/utils/bem'

export interface CellGroupProps {
  title: ReactNode
  description: ReactNode
  titleSlot: ReactNode
  descSlot: ReactNode
  className: string
  children?: ReactNode
}
const defaultProps = {
  title: '',
  description: '',
  className: '',
  titleSlot: null,
  descSlot: null,
} as CellGroupProps
export const CellGroup: FunctionComponent<Partial<CellGroupProps>> = (
  props
) => {
  const { children, className, title, description, titleSlot, descSlot } = {
    ...defaultProps,
    ...props,
  }
  const b = bem('cell-group')
  return (
    <div className={b(null, [className])}>
      {titleSlot || (
        <>{title ? <div className={b('title')}>{title}</div> : null}</>
      )}
      {descSlot || (
        <>
          {description ? (
            <div className={b('description')}>{description}</div>
          ) : null}
        </>
      )}

      <div className={b('wrap')}>{children}</div>
    </div>
  )
}

CellGroup.defaultProps = defaultProps
CellGroup.displayName = 'NutCellGroup'
