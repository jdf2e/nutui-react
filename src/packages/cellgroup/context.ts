import { createContext } from 'react'

const CellGroupContext = createContext<{ divider: boolean } | null>(null)

export default CellGroupContext
