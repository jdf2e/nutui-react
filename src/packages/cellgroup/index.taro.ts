import { CellGroup } from './cellgroup.taro'

export default CellGroup
