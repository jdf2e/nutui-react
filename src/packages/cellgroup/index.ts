import { CellGroup } from './cellgroup'

export default CellGroup
