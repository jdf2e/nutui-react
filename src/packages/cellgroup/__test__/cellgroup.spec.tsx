import * as React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import { Cell } from '../../cell/cell'
import { CellGroup } from '../cellgroup'

test('prop title description extra', () => {
  const { container } = render(
    <CellGroup title="链接 | 分组用法">
      <Cell title="链接" />
      <Cell title="URL 跳转" extra="https://jd.com" />
    </CellGroup>
  )
  expect(container.querySelectorAll('.nut-cell-group-title')[0].innerHTML).toBe(
    '链接 | 分组用法'
  )
  expect(container.querySelectorAll('.nut-cell').length).toEqual(2)
  expect(container).toMatchSnapshot()
})
