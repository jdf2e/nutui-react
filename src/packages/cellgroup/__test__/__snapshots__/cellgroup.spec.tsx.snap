// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prop title description extra 1`] = `
<div>
  <div
    class="nut-cell-group"
  >
    <div
      class="nut-cell-group__title"
    >
      链接 | 分组用法
    </div>
    <div
      class="nut-cell-group__wrap nut-cell-group__wrap--divider"
    >
      <div
        class="nut-cell"
        style="border-radius: 6px; align-items: flex-start;"
      >
        <div
          class="nut-cell__left"
        >
          <div
            class="nut-cell__title"
          >
            链接
          </div>
        </div>
        <div
          class="nut-cell__divider"
        />
      </div>
      <div
        class="nut-cell"
        style="border-radius: 6px; align-items: flex-start;"
      >
        <div
          class="nut-cell__left"
        >
          <div
            class="nut-cell__title"
          >
            URL 跳转
          </div>
        </div>
        <div
          class="nut-cell__extra"
        >
          https://jd.com
        </div>
        <div
          class="nut-cell__divider"
        />
      </div>
    </div>
  </div>
</div>
`;
